PIL/ImageFile.py

```diff
@@ -204,11 +206,11 @@ def load(self):
                 while True:
                     try:
                         s = read(self.decodermaxblock)
-                    except IndexError as ie:  # truncated png/gif
+                    except (IndexError, struct.error):  # truncated png/gif
                         if LOAD_TRUNCATED_IMAGES:
                             break
                         else:
-                            raise IndexError(ie)
+                            raise IOError("image file is truncated")
```

Comment:
I've replaced actual error with text because it is also text next in this function and because there was only one exception type before (IndexError), but for now there are two: `IndexError` and `struct.error`.